pravega/pravega
# Issue 600: Avoid stalling controller during event processor initialization

**Change log description**
Made event processor initialization asynchronous, so that controller is not stalled during this process. Moreover event processor initialization can take arbitrarily long at bootstrap, as it waits for pravega host to be available for creating commit and abort streams.

**Purpose of the change**
Fixes issue #600 

**What the code does**
1. Made controller event processor initialization asynchronous.
2. Made stream transaction task processor wait until commit and abort streams are created.

**How to verify it**
Controller bootstrap test case. It starts controller before starting host, and performs following operations
1. Create scope -- succeeds
2. Create stream -- waits for host to be up
3. Create txn -- fails because txn processing task modul is not yet initialized, it is waiting for commit/abort streams and reader groups to be created.

Then it starts host and does the following
1. Ensure that the previous create stream operation has succeeded
2. Create a new txn -- succeeds
